akka-actor/src/main/resources/reference.conf

```diff
@@ -350,6 +350,63 @@ akka {
+        parallelism-factor = 0.8
+
+        # Max number of threads to cap factor-based parallelism number to.
+        parallelism-max = 64
+
+        # Each worker in the pool uses a separate bounded MPSC queue. This value
+        # indicates the upper bound of the queue. Whenever an attempt to enqueue
+        # a task is made and the queue does not have capacity to accomodate
+        # the task, the rejection handler created by the factory specified
+        # in "rejection-handler-factory" is invoked.
+        task-queue-size = 512
+
+        # FQCN of the Rejection handler used in the pool.
+        # Must have an empty public constructor constructor with and needs
+        # to implement akka.actor.affinity.RejectionHandlerFactory.
+        rejection-handler-factory = "akka.dispatch.affinity.DefaultRejectionHandlerFactory"
+
+        # Level of CPU time used, on a scale between 1 and 10, during backoff/idle.
+        # The tradeoff is that to have low latency more CPU time must be used to be
+        # able to react quickly on incoming messages or send as fast as possible after
+        # backoff backpressure.
+        # Level 1 strongly prefer low CPU consumption over low latency.
+        # Level 10 strongly prefer low latency over low CPU consumption.
+        idle-cpu-level = 5
+
+        # Internally the AffinityPool uses two methods to determine which task
+        # queue to allocate a Runnable to:
+        # - map based - maintains a round robin counter and a map of Runnable
+        # hashcodes to queues that they have been associated with. This ensures
+        # maximum fairness in terms of work distribution, meaning that each worker
+        # will get approximately equal amount of mailboxes to execute. This is suitable
+        # in cases where we have a small number of actors that will be scheduled on
+        # the pool and we want to ensure the maximum possible utilization of the
+        # available threads.
+        # - hash based - the task - queue in which the runnable should go is determined
+        # by using an uniformly distributed int to int hash function which uses the
+        # hash code of the Runnable as an input. This is preferred in situations where we
+        # have enough number of distinct actors to ensure statistically uniform
+        # distribution of work across threads or we are ready to sacrifice the
+        # former for the added benefit of avoiding map look-ups.
+        #
+        # The value serves as a threshold which determines the point at which the
+        # pool switches from the first to the second work distribution schemes.
+        # For example, if the value is set to 2048, the pool can observe up to
+        # 2048 unique actors and schedule their mailboxes using the map based
+        # approach. Once this number is reached the pool switches to hash based
+        # task distribution mode.
+        fair-work-distribution-threshold = 2048
```

Comment:
I think it would be better to have a "task-queue-selector" FQCN. Then it would be possible to experiment with different hashing functions without hardcoding the CHM-based solution. @patriknw, wdyt?